core/src/main/java/com/redhat/lightblue/client/LightblueClient.java

```diff
@@ -1,18 +1,22 @@
 package com.redhat.lightblue.client;
 
+import com.redhat.lightblue.client.request.AbstractBulkLightblueRequest;
 import com.redhat.lightblue.client.request.AbstractLightblueDataRequest;
 import com.redhat.lightblue.client.request.LightblueRequest;
+import com.redhat.lightblue.client.response.BulkLightblueResponse;
 import com.redhat.lightblue.client.response.LightblueException;
 import com.redhat.lightblue.client.response.LightblueResponse;
 
 public interface LightblueClient {
 
-    Locking getLocking(String domain);
-    
-    LightblueResponse metadata(LightblueRequest lightblueRequest);
+	Locking getLocking(String domain);
 
-    LightblueResponse data(LightblueRequest lightblueRequest) throws LightblueException;
+	LightblueResponse metadata(LightblueRequest lightblueRequest);
 
-    <T> T data(AbstractLightblueDataRequest lightblueRequest, Class<T> type) throws LightblueException;
+	LightblueResponse data(LightblueRequest lightblueRequest) throws LightblueException;
+
+	BulkLightblueResponse dataBulk(AbstractBulkLightblueRequest<AbstractLightblueDataRequest> request) throws LightblueException;
```

Comment:
At least make it bulkData, instead of dataBulk

On Wed, Sep 9, 2015 at 7:22 AM, Brandon Vulaj <notifications@github.com>
wrote:

> In core/src/main/java/com/redhat/lightblue/client/LightblueClient.java
> <https://github.com/lightblue-platform/lightblue-client/pull/156#discussion_r39040631>
> :
>
> >
> > -    <T> T data(AbstractLightblueDataRequest lightblueRequest, Class<T> type) throws LightblueException;
> > +	LightblueResponse data(LightblueRequest lightblueRequest) throws LightblueException;
> > +
> > +	BulkLightblueResponse dataBulk(AbstractBulkLightblueRequest<AbstractLightblueDataRequest> request) throws LightblueException;
>
> @bserdar <https://github.com/bserdar> I changed this since method
> selection was somewhat ambigious, since AbstractBulkLightblueRequest
> implements LightblueRequest. I'll take a look at changing it back to data()
> and fixing any ambiguities in its use.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lightblue-platform/lightblue-client/pull/156/files#r39040631>
> .
>